src/modules/world/newMap/champAttacks.js

```diff
@@ -0,0 +1,74 @@
+import { cdn } from '../../system/system';
+import getElementsByTagName from '../../common/getElementsByTagName';
+import on from '../../common/on';
+import querySelector from '../../common/querySelector';
+import querySelectorArray from '../../common/querySelectorArray';
+import { sendEvent } from '../../support/fshGa';
+
+const creatureTypeIndex = ['NORMAL', 'CHAMPION', 'ELITE', 'SUPER ELITE', 'TITAN', 'LEGENDARY'];
+
+function getAttack(creatureElement) {
+  return querySelector('.verb.attack', creatureElement);
+}
+
+function getCreatures(creatureType) {
+  const index = creatureTypeIndex.indexOf(creatureType.toUpperCase());
+  return querySelectorArray(
+    `#actionList .actionListItem.creature.creature-${index}`,
+  );
+}
+
+function showChampAttack(toggle) {
+  const normalCreatures = ['LEGENDARY', 'NORMAL'].flatMap((e) => getCreatures(e));
+  const championCreatures = getCreatures('CHAMPION');
+
+  const [g1, g2] = toggle
+    ? [normalCreatures, championCreatures]
+    : [championCreatures, normalCreatures];
+
+  const attackIconPrefix = `url("${cdn}ui/world/icon_action_attack_`;
+
+  g1.map(getAttack)
+    .filter((e) => e !== null)
+    .forEach((e) => {
+      e.classList.remove('attack-1', 'attack-2', 'attack-3', 'attack-4',
+        'attack-5', 'attack-6', 'attack-7', 'attack-8');
+      e.style.backgroundImage = '';
+    });
+  g2.splice(0, 8)
+    .map(getAttack)
+    .forEach((e, i) => {
+      if (e !== null) {
```

Comment:
this 'if' should be a filter as above.
in fact .map(getAttack).filter(identity) could be combined into a function.
identity is (x) => x
the filter can operate on truthy.

Reply:
If I filter before the foreach, I'd lost the relative location of each monster. Otherwise, when the first monster is attacked, the hotkey numbers for all the others would shift up by one.

I could use map to keep the index by making a two dimensional array. Something like:
`g2.map((e,i) => [i, getAttack(e)]).filter((e) => e[1])`

Or I could use `compareDocumentPosition` with the first element of g2. I'd guess that'd be needless expensive though.

Reply:
good point, i see what you mean.

i am wondering if it would be worthwhile to remember the class and background.
haven't thought about it a lot. issue below seems to need state being stored.

stick with the 'if' for now.

Reply:
I think in the future it'd make sense to outright replace the action menu with svelte. It would be easier to store state changes, and then some of the additional components would be easier to integrate as well (i.e. titan stats). That's a bigger project though.